CacheLoaderTest and CacheWriterTest test failures

_master_ (commit 9f6027f5a5fcc1b8e0fa39821d0ea5c77d1cb610)

Failed on Sonar build : https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-sonar/261/testReport/

Failed test cases : 
```
com.hazelcast.client.cache.jsr.CacheLoaderTest.shouldPropagateExceptionUsingLoadAll
com.hazelcast.client.cache.jsr.CacheLoaderWithoutReadThroughTest.shouldPropagateExceptionUsingLoadAll
com.hazelcast.client.cache.jsr.CacheWriterTest.shouldWriteThroughRemoveAll_partialSuccess
com.hazelcast.client.cache.jsr.CacheWriterTest.shouldWrapWriterExceptions
com.hazelcast.client.cache.jsr.CacheWriterTest.shouldWriteThroughRemoveAllSpecific_partialSuccess
```
Stacktraces are like : 

```
org.junit.runners.model.TestTimedOutException: test timed out after 300000 milliseconds
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:304)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:160)
	at com.hazelcast.client.cache.impl.AbstractClientCacheProxyBase.invoke(AbstractClientCacheProxyBase.java:212)
	at com.hazelcast.client.cache.impl.AbstractClientInternalCacheProxy.removeAllKeysInternal(AbstractClientInternalCacheProxy.java:559)
	at com.hazelcast.client.cache.impl.ClientCacheProxy.removeAll(ClientCacheProxy.java:279)
	at org.jsr107.tck.integration.CacheWriterTest.shouldWriteThroughRemoveAllSpecific_partialSuccess(CacheWriterTest.java:1033)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standart output's like :
```
Started Running Test: shouldWriteThroughRemoveAllSpecific_partialSuccess
22:00:41,035  INFO || - [CacheService] hz._hzInstance_11_dev.partition-operation.thread-69 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Added cache config: CacheConfig{name='cache-writer-test', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, wanReplicationRef=null}
22:00:41,563 ERROR || - [Invocation] hz._hzInstance_11_dev.async.thread-69 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Failed asynchronous execution of execution callback: com.hazelcast.spi.impl.operationservice.impl.InvokeOnPartitionsAsync$FirstAttemptExecutionCallback@371f4a48for call Invocation{op=com.hazelcast.spi.impl.operationservice.impl.operations.PartitionIteratingOperation{serviceName='hz:impl:cacheService', identityHash=1136818413, partitionId=-1, replicaIndex=0, callId=-868, invocationTime=1540245641242 (2018-10-22 22:00:41.242), waitTimeout=-1, callTimeout=120000, operationFactory=OperationFactoryWrapper{opFactory=com.hazelcast.cache.impl.operation.CacheRemoveAllOperationFactory@39d8b2db, uuid='03181d4c-3272-4373-ad12-7584f5a44f32'}}, tryCount=10, tryPauseMillis=300, invokeCount=1, callTimeoutMillis=120000, firstInvocationTimeMs=1540245641242, firstInvocationTime='2018-10-22 22:00:41.242', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 00:00:00.000', target=[127.0.0.1]:5701, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
javax.cache.integration.CacheWriterException: Exception in CacheWriter during deleteAll
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.deleteAllCacheEntry(AbstractCacheRecordStore.java:1045) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.removeAll(AbstractCacheRecordStore.java:1675) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.operation.CacheRemoveAllOperation.run(CacheRemoveAllOperation.java:84) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.Operation.call(Operation.java:170) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:208) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:197) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:147) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:177) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:131) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
Caused by: javax.cache.CacheException: simulated delete failure for key 1
	at org.jsr107.tck.integration.BatchPartialSuccessRecordingClassWriter.deleteAll(BatchPartialSuccessRecordingClassWriter.java:107) ~[cache-tests-1.1.0.jar:?]
	at org.jsr107.tck.integration.CacheWriterServer$DeleteAllOperationHandler.onProcess(CacheWriterServer.java:95) ~[cache-tests-1.1.0.jar:?]
	at org.jsr107.tck.support.Server$ClientConnection.run(Server.java:310) ~[cache-tests-1.1.0.jar:?]
22:05:41,058  INFO || - [CacheService] hz._hzInstance_11_dev.client.thread-45 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Removed cache config: CacheConfig{name='cache-writer-test', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, wanReplicationRef=null}
```
```
Finished Running Test: shouldNotLoadUsingGetAndPut in 0.004 seconds.
Started Running Test: shouldPropagateExceptionUsingLoadAll
22:05:59,184  INFO || - [CacheService] hz._hzInstance_20_dev.partition-operation.thread-63 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Added cache config: CacheConfig{name='cache-loader-test', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, wanReplicationRef=null}
22:05:59,509 ERROR || - [Invocation] hz._hzInstance_20_dev.async.thread-63 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Failed asynchronous execution of execution callback: com.hazelcast.spi.impl.operationservice.impl.InvokeOnPartitionsAsync$FirstAttemptExecutionCallback@36ff5050for call Invocation{op=com.hazelcast.spi.impl.operationservice.impl.operations.PartitionIteratingOperation{serviceName='hz:impl:cacheService', identityHash=1312634646, partitionId=-1, replicaIndex=0, callId=-132, invocationTime=1540245959186 (2018-10-22 22:05:59.186), waitTimeout=-1, callTimeout=120000, operationFactory=OperationFactoryWrapper{opFactory=com.hazelcast.cache.impl.operation.CacheLoadAllOperationFactory@3838b8ff, uuid='472f06a7-f4de-4c89-810a-6c9657215c66'}}, tryCount=10, tryPauseMillis=300, invokeCount=1, callTimeoutMillis=120000, firstInvocationTimeMs=1540245959186, firstInvocationTime='2018-10-22 22:05:59.186', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 00:00:00.000', target=[127.0.0.1]:5701, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
javax.cache.integration.CacheLoaderException: Exception in CacheLoader during loadAll
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.loadAllCacheEntry(AbstractCacheRecordStore.java:1070) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.loadAll(AbstractCacheRecordStore.java:1705) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.operation.CacheLoadAllOperation.run(CacheLoadAllOperation.java:98) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.Operation.call(Operation.java:170) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:208) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:197) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:147) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:177) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:131) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
Caused by: java.lang.UnsupportedOperationException
	at org.jsr107.tck.integration.FailingCacheLoader.loadAll(FailingCacheLoader.java:32) ~[cache-tests-1.1.0.jar:?]
	at org.jsr107.tck.integration.CacheLoaderServer$LoadAllOperationHandler.onProcess(CacheLoaderServer.java:98) ~[cache-tests-1.1.0.jar:?]
	at org.jsr107.tck.support.Server$ClientConnection.run(Server.java:310) ~[cache-tests-1.1.0.jar:?]
22:11:09,255  INFO || - [CacheService] hz._hzInstance_20_dev.client.thread-1 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Removed cache config: CacheConfig{name='cache-loader-test', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, wanReplicationRef=null}
```
```
21:55:38,768 ERROR || - [CachePutOperation] hz._hzInstance_11_dev.partition-operation.thread-32 - [127.0.0.1]:5701 [dev] [3.12-SNAPSHOT] Exception in CacheWriter during write
javax.cache.integration.CacheWriterException: Exception in CacheWriter during write
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.writeThroughCache(AbstractCacheRecordStore.java:1005) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.createRecord(AbstractCacheRecordStore.java:675) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.createRecordWithExpiry(AbstractCacheRecordStore.java:699) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.createRecordWithExpiry(AbstractCacheRecordStore.java:728) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.put(AbstractCacheRecordStore.java:1263) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.AbstractCacheRecordStore.put(AbstractCacheRecordStore.java:1294) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.cache.impl.operation.CachePutOperation.run(CachePutOperation.java:61) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.Operation.call(Operation.java:170) ~[hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:208) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:197) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:407) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:434) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:586) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:571) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:530) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:220) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:60) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:67) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:123) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.doRun(AbstractMessageTask.java:111) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:101) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:161) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:159) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:127) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110) [hazelcast-3.12-SNAPSHOT.jar:3.12-SNAPSHOT]
Caused by: java.lang.UnsupportedOperationException
	at org.jsr107.tck.integration.FailingCacheWriter.write(FailingCacheWriter.java:24) ~[cache-tests-1.1.0.jar:?]
	at org.jsr107.tck.integration.CacheWriterServer$WriteOperationHandler.onProcess(CacheWriterServer.java:274) ~[cache-tests-1.1.0.jar:?]
	at org.jsr107.tck.support.Server$ClientConnection.run(Server.java:310) `~[cache-tests-1.1.0.jar:?]
```

--- a/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractAllPartitionsMessageTask.java
+++ b/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractAllPartitionsMessageTask.java
@@ -51,6 +51,10 @@ public abstract class AbstractAllPartitionsMessageTask<P> extends AbstractMessag
 
     @Override
     public final void onResponse(Map<Integer, Object> map) {
-        sendResponse(reduce(map));
+        try {
+            sendResponse(reduce(map));
+        } catch (Exception e) {
+            handleProcessingFailure(e);
+        }
     }
 }

